### PR TITLE
Added Faker.Date for generating random dates

### PR DESCRIPTION
**Pull request by tobyhinloopen**

### CODE REVIEW

**Comment by tobyhinloopen** [comment]
Build is failing but the failures are not caused by my code changes. In addition, since Elixir's Date module only exists in version >=1.3.0, i've added a version check where Faker.Date is only defined for elixir versions >= 1.3.0

**Comment by tobyhinloopen** [comment]
Rebased & squashed commits.

File: lib/faker/date.ex [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    Functions for generating dates
+    """
+
+    @seconds_per_day 86400
+
+    @doc """
+    Returns a random date of birth for a person aged between ~12 to ~99
+    """
+    @spec date_of_birth() :: Date.t
+    def date_of_birth do
```
Comment (igas) [inline lib/faker/date.ex]: What do you think if we'll add 3 variations?

- default 18..99
- 16 - exact
- 18..27 - in between

Comment (tobyhinloopen) [inline lib/faker/date.ex]: I like that :) Fixed

**Comment by tobyhinloopen** [comment]
I checked the docs (and added a minor fix); I've added an entry in USAGE.md and CHANGELOG.md

**Comment by tobyhinloopen** [comment]
& figured out how to rebase from a forked repo

**Comment by igas** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:

**Pull request merged**